src/lib.rs

```diff
@@ -598,7 +601,8 @@ impl IntoIter {
         // Make room for another open file descriptor if we've hit the max.
         if self.stack_list.len() - self.oldest_opened == self.opts.max_open {
             self.stack_list[self.oldest_opened].close();
-            self.oldest_opened = self.oldest_opened.checked_add(1).unwrap();
+            // Unwrap is safe here as long as we assume that `self.opts.max_open` is never 0
```

Comment:
> Upon further thought, is there a reason this was written as a checked_add rather than just using + 1?

Because `+ 1` will silently overflow, which would be a silent bug.

But, you make a good point about the subtraction. Perhaps the right thing here is to change the `if` to `self.stack_list.len().checked_sub(self.oldest_opened).unwrap() == self.ops.max_open` and keep the `checked_add`. The comment can simply state that `self.oldest_opened` is guaranteed to never be greater than `self.stack_list.len()`, which implies that the subtraction won't underflow and that adding `1` will never overflow.